test/jdk/java/text/Format/NumberFormat/CurrencyFormat.java

```diff
@@ -42,127 +46,140 @@
 import java.text.DecimalFormatSymbols;
 import java.text.NumberFormat;
 import java.text.SimpleDateFormat;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class CurrencyFormat {
 
-    private static boolean isCompat;
+    // Expected data is switched depending on COMPAT or CLDR
+    // currencySymbolsTest() is only ran for COMPAT
+    private static final boolean isCompat =
+            "COMPAT".equals(System.getProperty("java.locale.providers"));
```

Comment:
This check could be moved into `currencySymbolsTest()`

Reply:
I left it as a static declaration as although it is used in `currencySymbolsTest()`, it is also used in the data provider of `CurrencyFormatTest()`.

Reply:
Missed that. Thanks.